tests/test_mentions.py

```
@@ -0,0 +1,506 @@
+        ("email user@domain.com", []),  # email — no match
+        ("no mentions here", []),
+        ("@alice-bob", ["alice-bob"]),  # hyphens allowed
+        ("@alice_bob", ["alice_bob"]),  # underscores allowed
+        ("@alice123", ["alice123"]),  # digits allowed
+        (".@alice", []),  # preceded by dot — no match
+        ("x@alice", []),  # preceded by letter — no match
+        ("1@alice", []),  # preceded by digit — no match
+        ("(@alice)", ["alice"]),  # parens ok
+        ("@alice.", ["alice"]),  # trailing dot ok
+        ("@alice ping @alice", ["alice"]),  # deduplication — same user only once
+    ]
```

Ruff (0.15.11)

[warning] Mutable default value for class attribute

(RUF012)